Change home page UI (#1875)
plone/documentation: docs/index.md

This is the community-maintained documentation for the Plone content management system.

:::::{grid} 1 1 2 2
:gutter: 3 3 4 5

::::{grid-item-card} 
:link: install/index
:link-type: doc
{octicon}`zap;1.5em;sd-text-info`
Get started
^^^
This part of the documentation helps you find the best way to get started with Plone, depending on what you want to do.
+++
```{button-ref} {doc} install/index
:color: primary
:expand:
Go to Get started
``` 
::::

::::{grid-item-card} 
:link: user-guide/index
:link-type: doc
{octicon}`person;1.5em;sd-text-info`
User guide 
^^^

> This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and common tasks for users and administrators.

Reviewer: ```suggestion
This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and other common tasks for site editors and administrators.
```

+++
```{button-ref} {doc} user-guide/index
:color: primary
:expand:
Go to User guide
```
::::

::::{grid-item-card}
:link: developer-guide/index
:link-type: doc
{octicon}`person-add;1.5em;sd-text-info`
Developer guide
^^^

> This part of the documentation describes how to develop a Plone site.

Reviewer: ```suggestion
This part of the documentation describes how to develop a Plone project.
```

+++
```{button-ref} {doc} developer-guide/index 
:color: primary
:expand:
Go to Developer guide
```
::::

::::{grid-item-card} 
:link: reference-guide/index
:link-type: doc
{octicon}`bookmark;1.5em;sd-text-info`
Reference guide 
^^^

> This part of the documentation provides explanation of APIs, functions, modules, and objects included in Plone, describing what they are and what they do.

Reviewer: ```suggestion
This part of the documentation describes the APIs, functions, modules, and objects included in Plone.
```

+++
```{button-ref} {doc} reference-guide/index 
:color: primary
:expand:
Go to Reference guide
```
::::

::::{grid-item-card} 
:link: conceptual-guides/index
:link-type: doc
{octicon}`light-bulb;1.5em;sd-text-info`
Conceptual guides 
^^^
This part of the documentation provides explanation of concepts to deepen and broaden your understanding of Plone.
+++
```{button-ref} {doc} conceptual-guides/index 
:color: primary
:expand:
Go to Conceptual guides
```
::::

::::{grid-item-card} 
:link: contributing/index
:link-type: doc

> {octicon}`book;1.5em;sd-text-info`

Reviewer: The book icon represents documentation, whereas the pull request icon relates better to making a contribution.

```suggestion
{octicon}`git-pull-request;1.5em;sd-text-info`
```

This is the community-maintained documentation for the Plone content management system.

:::::{grid} 1 1 2 2
:gutter: 3 3 4 5

::::{grid-item-card} 
:link: https://6.docs.plone.org/contributing/index.html
{octicon}`book;1.5em;sd-text-info`
Contributing guide
^^^
This part of the documentation describes how to contribute to Plone, including all its projects and repositories under the Plone GitHub organization.
+++
```{button-link} https://6.docs.plone.org/contributing/index.html
:color: primary
:expand:
Go to Contributing guide
```

> ::::

Reviewer: Let's move the Contributing guide card to the last position.

::::{grid-item-card} 
:link: https://6.docs.plone.org/install/index.html
{octicon}`zap;1.5em;sd-text-info`
Get started
^^^
This part of the documentation helps you find the best way to get started with Plone, depending on what you want to do.
+++
```{button-link} https://6.docs.plone.org/install/index.html
:color: primary
:expand:
Go to Get started
```
::::

> 

Reviewer: I think Get Started should be at the start. Let's move this to the first card.

I think we should also add a "User guide" card as the second card. We will need a new page as well, `user-guide/index.md`. It should include a description and two links.

1. https://6.docs.plone.org/volto/user-manual/index.html
2. [Content Editing for Plone](https://training.plone.org/content-editing/index.html).

::::{grid-item-card}
:link: https://6.docs.plone.org/developer-guide/index.html
{octicon}`person;1.5em;sd-text-info`
Developer guide
^^^
This part of the documentation provides information for how to develop in Plone.
+++
```{button-link} https://6.docs.plone.org/developer-guide/index.html
:color: primary
:expand:
Go to Developer guide
```

> ::::

Reviewer: Let's move the Developer guide card to the third position.

Let's add a card for the Reference guide card to the fourth position.

::::{grid-item-card} 
:link: https://6.docs.plone.org/conceptual-guides/index.html
{octicon}`light-bulb;1.5em;sd-text-info`
Conceptual guides 
^^^
This part of the documentation provides explanation of concepts to deepen and broaden your understanding of Plone.
+++
```{button-link} https://6.docs.plone.org/conceptual-guides/index.html
:color: primary
:expand:
Go to Conceptual guides
```
::::

> :::::

Reviewer: Let's move the Conceptual guides card to the fifth position.

This is the community-maintained documentation for the Plone content management system.

:::::{grid} 1 1 2 2
:gutter: 3 3 4 5

::::{grid-item-card} 
:link: https://6.docs.plone.org/install/index.html
{octicon}`zap;1.5em;sd-text-info`
Get started
^^^
This part of the documentation helps you find the best way to get started with Plone, depending on what you want to do.
+++
```{button-link} https://6.docs.plone.org/install/index.html
:color: primary
:expand:
Go to Get started
```
::::

::::{grid-item-card} 

> :link: http://127.0.0.1:8050/user-guide/index.html

Reviewer 2: (This feedback is for all the links on this page.)

We can't use a hardcoded link to localhost or the published docs! They aren't correct when trying to preview this page on https://plone6--1875.org.readthedocs.build/

@stevepiercy Is it possible to use sphinx-style links here?

Reviewer: All links to a page to internal documentation should use the following pattern.

```
{doc}`path/to/page`
```

The page should not include the file suffix of `.md`. This should render to HTML with correct relative URLs.

Reviewer: See reference https://6.docs.plone.org/contributing/documentation/myst-reference.html#cross-references

Author: Hey @stevepiercy, I tried using your method for linking, but the problem is that Sphinx Design does not support the {doc} syntax inside it. The link part may work, but the button's styling disappears. Could you try it locally and show me how it works?

Reviewer: Ah, that's right, there's a special syntax. See https://sphinx-design.readthedocs.io/en/latest/cards.html#clickable-cards.

```
:::{card} Clickable Card (internal)
:link: cards-clickable
:link-type: ref
:link-alt: clickable cards

The entire card can be clicked to navigate to the `cards-clickable` reference target.
:::
```

So for a valid link to a page, you'd do something like this.

```
:link: path/to/page
:link-type: doc
```

{octicon}`person;1.5em;sd-text-info`
User guide 
^^^
This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and common tasks for users and administrators.
+++
```{button-link} http://127.0.0.1:8050/user-guide/index.html
:color: primary
:expand:
Go to User guide
```
::::

> 

Reviewer 2: @stevepiercy Seems to me that the Admin guide should also be mentioned at this same level.

Reviewer: I regret the distinction of an Admin guide from Developer guide, and I would like to merge them. Admin, Develop, and Deploy have a lot of overlapping requisite knowledge. This also makes it easier to have related content in a single guide and makes it easier to have separation of content along the Diátaxis framework.

Reviewer 2: @stevepiercy I sort of see your point.

Let me express my main concern: The admin guide explains how to run Plone, even if you aren't extending it. "Developer guide" doesn't strike me as an intuitive place to look for that.

Maybe it can work as long as we make it clear that it's for people who are developing a Plone site (not just developing code for a Plone site) and that it includes sysadmin/devops topics as well. The current blurb for the Develop guide card doesn't make this clear. ("This part of the documentation provides information for how to develop in Plone.")

Reviewer: > The admin guide explains how to run Plone, even if you aren't extending it. 

@davisagli it provides that, and a lot more. 😞 

![Screenshot 2025-03-03 at 9 40 44 PM](https://github.com/user-attachments/assets/31d8cf93-346c-485a-ac61-963d0cb1da4d)

How about this for Develop guide?

"This part of the documentation describes how to install, extend, deploy, maintain, and other technical topics to develop a Plone site."

Reviewer 2: @stevepiercy Well, that certainly covers it, but is a bit verbose. Honestly just "This part of the documentation describes how to develop a Plone site." could work. At first glance that might not look much different from what's already proposed in this PR, but I think "develop a Plone site" expresses a much clearer goal than "develop in Plone"

I don't mean to bikeshed. Let's try something and we'll find out where people still need help with wayfinding. It's already plenty better than the current home page.

Author: @stevepiercy @davisagli What's the final decision? What am I supposed to do?

Reviewer 2: @Manas-Kenge In the end it's a small change: Ignore my suggestion to add the admin guide here. Change the description of the Developer guide to "This part of the documentation describes how to develop a Plone site."

::::{grid-item-card}
:link: https://6.docs.plone.org/developer-guide/index.html
{octicon}`person;1.5em;sd-text-info`
Developer guide
^^^
This part of the documentation provides information for how to develop in Plone.
+++
```{button-link} https://6.docs.plone.org/developer-guide/index.html
:color: primary
:expand:
Go to Developer guide
```
::::

::::{grid-item-card} 
:link: http://127.0.0.1:8050/reference-guide/index.html
{octicon}`bookmark;1.5em;sd-text-info`
Reference guide 
^^^
This part of the documentation provides explanation of APIs, functions, modules, and objects included in Plone, describing what they are and what they do.
+++
```{button-link} http://127.0.0.1:8050/reference-guide/index.html
:color: primary
:expand:
Go to Reference guide
```
::::

::::{grid-item-card} 
:link: https://6.docs.plone.org/conceptual-guides/index.html
{octicon}`light-bulb;1.5em;sd-text-info`

> Conceptual guides 

Reviewer 2: I would put Conceptual guides before Developer guide. It's probably a good idea to understand some concepts before you dive into developing.

Reviewer: For some conceptual guides, like choosing a user interface or compare buildout and pip, I'd agree. In this case, a how-to guide includes a link to the relevant conceptual guide for deeper understanding if the reader wants to go there. Others require advanced technical knowledge to understand, or describe how sausage is made, and for those I'd disagree.

When I develop, I seek how-tos first, API references second, source code third, and as a last resort conceptual guides. I think that's what most developers do, try stuff out, and when it doesn't work, dig deeper and deeper.

Reviewer 2: @stevepiercy You may be right and I don't feel strongly about this. I'll resolve the comment, and we can always revisit it later.

^^^
This part of the documentation provides explanation of concepts to deepen and broaden your understanding of Plone.
+++
```{button-link} https://6.docs.plone.org/conceptual-guides/index.html
:color: primary
:expand:
Go to Conceptual guides
```
::::

::::{grid-item-card} 
:link: https://6.docs.plone.org/contributing/index.html
{octicon}`book;1.5em;sd-text-info`

> Contributing guide

Reviewer 2: ```suggestion
Contributor guide
```

^^^
This part of the documentation describes how to contribute to Plone, including all its projects and repositories under the Plone GitHub organization.
+++
```{button-link} https://6.docs.plone.org/contributing/index.html
:color: primary
:expand:

> Go to Contributing guide

Reviewer 2: ```suggestion
Go to Contributor guide
```
